public/app/plugins/datasource/loki/components/LokiCheatSheet.tsx

```diff
@@ -1,43 +1,127 @@
+import { ExploreStartPageProps, DataQuery } from '@grafana/ui';
+import LokiLanguageProvider from '../language_provider';
+
+// const CHEAT_SHEET_ITEMS = [
+//   {
+//     title: 'See your logs',
+//     label: '',
+//   },
+//   {
+//     title: 'Logs from a "job"',
+//     expression: '{job="default/prometheus"}',
+//     label: 'Returns all log lines emitted by instances of this job.',
+//   },
+//   {
+//     title: 'Combine stream selectors',
+//     expression: '{app="cassandra",namespace="prod"}',
+//     label: 'Returns all log lines from streams that have both labels.',
+//   },
+//   {
+//     title: 'Search for text',
+//     expression: '{app="cassandra"} (duration|latency)\\s*(=|is|of)\\s*[\\d\\.]+',
+//     label: 'Add a regular expression after the selector to filter for.',
+//   },
+// ];
+
+const DEFAULT_EXAMPLES = ['{job="default/prometheus"}'];
+const PREFERRED_LABELS = ['job', 'app', 'k8s_app'];
+
+export default class LokiCheatSheet extends PureComponent<ExploreStartPageProps, { userExamples: string[] }> {
+  userLabelTimer: NodeJS.Timeout;
+  state = {
+    userExamples: DEFAULT_EXAMPLES,
+  };
+
+  componentDidMount() {
+    this.scheduleUserLabelChecking();
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.userLabelTimer);
+  }
+
+  scheduleUserLabelChecking() {
+    this.userLabelTimer = setTimeout(this.checkUserLabels, 1000);
+  }
+
+  checkUserLabels = async () => {
+    // Set example from user labels
+    const provider: LokiLanguageProvider = this.props.datasource.languageProvider;
+    if (provider.started) {
+      const labels = provider.getLabelKeys() || [];
+      const preferredLabel = PREFERRED_LABELS.find(l => labels.includes(l));
+      if (preferredLabel) {
+        const values = await provider.getLabelValues(preferredLabel);
+        const userExamples = shuffle(values)
+          .slice(0, 3)
+          .map(value => `{${preferredLabel}="${value}"}`);
+        this.setState({ userExamples });
+      }
+    } else {
+      this.scheduleUserLabelChecking();
+    }
+  };
+
+  render() {
+    const { onClickExample } = this.props;
+    const { userExamples } = this.state;
+
+    return (
+      <div>
+        <h2>Loki Cheat Sheet</h2>
+        <div className="cheat-sheet-item">
+          <div className="cheat-sheet-item__title">See your logs</div>
+          <div className="cheat-sheet-item__label">Start by selecting a log stream from the Log labels selector.</div>
+          <div className="cheat-sheet-item__label">
+            Alternatively, you can write a stream selector into the query field:
+          </div>
+          <div className="cheat-sheet-item__expression">
+            <code>{'{job="default/prometheus"}'}</code>
+          </div>
+          {userExamples !== DEFAULT_EXAMPLES && userExamples.length > 0 ? (
+            <div>
+              <div className="cheat-sheet-item__label">Here are some example streams from your logs:</div>
+              {userExamples.map(example => (
```

Comment:
The clickable and non clickable labels look the same so it is easy to overlook that. Not sure though what would be a great visual cue here.

Reply:
Interestingly for Prometheus all seem clickable, even though they are generic and for me locally the queries there do not produce any data.

Reply:
I liked this style of presenting clickable commands in Kubernetes Basics tutorial:

<img width="1167" alt="Screenshot 2019-09-06 at 17 52 27" src="https://user-images.githubusercontent.com/895919/64441968-4111bd00-d0cf-11e9-86b3-b25fbd8efa14.png">

Reply:
Decided to go with no distinction between user data and fake examples.